.github/workflows/ci.yaml

```diff
@@ -0,0 +1,70 @@
+# workflow name shown in the Actions tab and PR checks
+name: ci-gha
+
+on:
+  pull_request:
+    branches:
+      - main
+
+  push:
+    branches:
+      - main
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - uses: ./.github/actions/setup
+      - name: Build with Gradle
+        run: ./gradlew test --no-daemon --debug --max-workers 2
+
+  it-test:
+    strategy:
+      # might be helpful to limit threads in local builds
+#      max-parallel: 2
+      matrix:
+        it: [it1, it2, it3, it4, it5, it6, it7, it8]
```

Comment:
See [comment](https://github.com/sonatype-nexus-community/scan-gradle-plugin/pull/169/files#r1815025908). `version3` reduced the number of integration tests to 6 I think.

Reply:
Make sense. Added a TODO to update after `version3` merge. [ba6d43a](https://github.com/sonatype-nexus-community/scan-gradle-plugin/pull/169/commits/ba6d43a16e641d9a2713440711d91ca3ce1370d2)